typescript/sdk/src/metadata/ChainMetadataManager.ts

```diff
@@ -55,8 +56,10 @@ export class ChainMetadataManager<MetaExt = {}> {
    * @throws if chain's name or domain/chain ID collide
    */
   addChain(metadata: ChainMetadata<MetaExt>): void {
-    if (!isValidChainMetadata(metadata))
+    if (!isValidChainMetadata(metadata)) {
+      ChainMetadataSchema.parse(metadata);
```

Comment:
Why add this parse line here? The isValid method also parses

Reply:
to see the error
I can remove

Reply:
If `parse` throws then why keep the throw line after it? I'm open to surfacing the error but I'd suggest we restructure this to avoid parsing twice / unreachable code. If you don't want to do that now feel free to make a ticket for me

Reply:
Should we just call parse, and then check that the result is an error?